Kustomize edit panics when Kustomization file begins with document separator

**Describe the bug**

I get a panic when running `kustomize edit set namespace mynamespace`.

**Files that can reproduce the issue**

kustomization.yaml

```yaml

---
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

namespace: freeipa
```

**Expected output**

No panic message and the `namespace` field updated to `mynamespace`.

**Actual output**

```raw
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile.(*kustomizationFile).parseCommentedFields(0xc0002e9140, 0xc000030c60, 0x5a, 0x60, 0x0, 0x0)
        /workspace/myClone/kustomize/commands/internal/kustfile/kustomizationfile.go:216 +0x7c5
sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile.(*kustomizationFile).Read(0xc0002e9140, 0xf66538, 0xc0002e9140, 0x0)
        /workspace/myClone/kustomize/commands/internal/kustfile/kustomizationfile.go:174 +0x14a
sigs.k8s.io/kustomize/kustomize/v4/commands/edit/set.(*setNamespaceOptions).RunSetNamespace(0xc000275940, 0xbcb760, 0xf66538, 0x1, 0x0)
        /workspace/myClone/kustomize/commands/edit/set/setnamespace.go:66 +0x65
sigs.k8s.io/kustomize/kustomize/v4/commands/edit/set.newCmdSetNamespace.func1(0xc000315080, 0xc000320420, 0x1, 0x1, 0x0, 0x0)
        /workspace/myClone/kustomize/commands/edit/set/setnamespace.go:41 +0xd0
github.com/spf13/cobra.(*Command).execute(0xc000315080, 0xc0003203e0, 0x1, 0x1, 0xc000315080, 0xc0003203e0)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc000308000, 0x0, 0xa12400, 0xc000030118)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        /workspace/myClone/kustomize/main.go:14 +0x2a
```

**Kustomize version**

```raw
kustomize version
{Version:kustomize/v4.2.0 GitCommit:d53a2ad45d04b0264bcee9e19879437d851cb778 BuildDate:2021-06-30T22:49:26Z GoOs:linux GoArch:amd64}
```

**Platform**

Linux

**Additional context**

Installed by:

```shell
curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash
```

--- a/kustomize/commands/internal/kustfile/kustomizationfile.go
+++ b/kustomize/commands/internal/kustfile/kustomizationfile.go
@@ -212,7 +212,7 @@ func (mf *kustomizationFile) parseCommentedFields(content []byte) error {
 			if matched {
 				mf.originalFields = append(mf.originalFields, &commentedField{field: field, comment: squash(comments)})
 				comments = [][]byte{}
-			} else if len(comments) > 0 {
+			} else if len(comments) > 0 && len(mf.originalFields) > 0 {
 				mf.originalFields[len(mf.originalFields)-1].appendComment(squash(comments))
 				comments = [][]byte{}
 			}

--- a/kustomize/commands/internal/kustfile/kustomizationfile_test.go
+++ b/kustomize/commands/internal/kustfile/kustomizationfile_test.go
@@ -356,6 +356,53 @@ kind: Kustomization
 	}
 }
 
+func TestCommentsWithDocumentSeperatorAtBeginning(t *testing.T) {
+	kustomizationContentWithComments := []byte(`
+
+    
+# Some comments
+# This is some comment we should preserve
+# don't delete it
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: mynamespace
+`)
+
+	expected := []byte(`
+
+    
+# Some comments
+# This is some comment we should preserve
+# don't delete it
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: mynamespace
+`)
+	fSys := filesys.MakeFsInMemory()
+	testutils_test.WriteTestKustomizationWith(
+		fSys, kustomizationContentWithComments)
+	mf, err := NewKustomizationFile(fSys)
+	if err != nil {
+		t.Fatalf("Unexpected Error: %v", err)
+	}
+
+	kustomization, err := mf.Read()
+	if err != nil {
+		t.Fatalf("Unexpected Error: %v", err)
+	}
+	if err = mf.Write(kustomization); err != nil {
+		t.Fatalf("Unexpected Error: %v", err)
+	}
+	bytes, _ := fSys.ReadFile(mf.path)
+
+	if diff := cmp.Diff(expected, bytes); diff != "" {
+		t.Errorf("Mismatch (-expected, +actual):\n%s", diff)
+	}
+}
+
 func TestUnknownFieldInKustomization(t *testing.T) {
 	kContent := []byte(`
 foo:

